test/pylib/tablets.py

```diff
@@ -42,6 +43,18 @@ async def get_all_tablet_replicas(manager: ManagerClient, server: ServerInfo, ke
         replicas=[(HostID(str(host)), shard) for (host, shard) in x.replicas]
     ) for x in rows]
 
+async def get_replica_count_by_host(manager: ManagerClient, server: ServerInfo, keyspace_name: str, table_name: str, is_view: bool = False) -> dict[HostID, int]:
+    """
+    Retrieves the count of tablet replicas per host for a given table.
+
+    :param server: server to query. Can be any live node.
+    """
+    result = defaultdict(int)
+    for tablet in await get_all_tablet_replicas(manager, server, keyspace_name, table_name, is_view):
+        for host, shard in tablet.replicas:
+            result[host] += 1
+    return result
+
 async def get_tablet_replicas(manager: ManagerClient, server: ServerInfo, keyspace_name: str, table_name: str, token: int) -> list[tuple[HostID, int]]:
     """
     Gets tablet replicas of the tablet which owns a given token of a given table.
```

Comment:
Some test cases may be missing:
- it's possible to bootstrap a node while a decommission request is paused,
- parallel removenode and decommission (not sure interesting).

But overall the tests seem very good and cover the problematic cases.